samples/nrf9160/serial_lte_modem/src/lwm2m/th_lwm2m_service.c

```diff
@@ -0,0 +1,727 @@
+		LOG_INF("Registration complete");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_REG_UPDATE_FAILURE:
+		LOG_ERR("Registration update failure!");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_REG_UPDATE_COMPLETE:
+		LOG_INF("Registration update complete");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_DEREGISTER_FAILURE:
+		LOG_ERR("Deregister failure!");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_DISCONNECT:
+		LOG_INF("Disconnected");
+		break;
+
+	default:
+		LOG_WRN("Unknown event");
+		break;
+	}
+
+	evt = client_event;
+	(void)inter_connect_notify(NOT_TYPE_LWM2M_RD, &evt, 1);
+}
+
+int do_lwm2m_connect(const u8_t *endpoint_name, u8_t length)
+{
+	int ret;
+
+	ret = lwm2m_setup();
+	if (ret < 0) {
+		LOG_ERR("Cannot setup LWM2M fields (%d)", ret);
+		return ret;
+	}
+
+	(void)memset(&client, 0x0, sizeof(client));
+
+	/* client.sec_obj_inst is 0 as a starting point */
+	lwm2m_rd_client_start(&client, endpoint_name, rd_client_event);
+	return 0;
+}
+
+int do_lwm2m_disconnect(void)
+{
+	lwm2m_rd_client_stop(&client, rd_client_event);
+	return 0;
+}
+
+int do_lwm2m_read_int(const u8_t *param, u8_t length, u8_t *output)
+{
+	u8_t int_type = *param;
+	char path[MAX_URI_LENGTH];
+	int ret = -EINVAL;
+
+	if (length > 1) {
+		memcpy(path, (param+1), length-1);
+	} else {
+		return ret;
+	}
+	switch (int_type) {
+	case LWM2M_INT_TYPE_BOOLEAN: {
+		bool value;
+
+		ret = lwm2m_engine_get_bool(path, &value);
+		if (ret == 0) {
+			*output = value;
+			ret = 1;
+		}
+	} break;
+	case LWM2M_INT_TYPE_UINT8: {
+		u8_t value;
+
+		ret = lwm2m_engine_get_u8(path, &value);
+		if (ret == 0) {
+			*output = value;
+			ret = 1;
+		}
+	} break;
+	case LWM2M_INT_TYPE_UINT16: {
+		u16_t value;
+
+		ret = lwm2m_engine_get_u16(path, &value);
+		if (ret == 0) {
+			memcpy(output, &value, 2);
+			ret = 2;
+		}
+	} break;
+	case LWM2M_INT_TYPE_UINT32: {
+		u32_t value;
+
+		ret = lwm2m_engine_get_u32(path, &value);
+		if (ret == 0) {
+			memcpy(output, &value, 4);
+			ret = 4;
+		}
+	} break;
+	case LWM2M_INT_TYPE_UINT64: {
+		u64_t value;
+
+		ret = lwm2m_engine_get_u64(path, &value);
+		if (ret == 0) {
+			memcpy(output, &value, 8);
+			ret = 8;
+		}
+	} break;
+	case LWM2M_INT_TYPE_INT8: {
+		s8_t value;
+
+		ret = lwm2m_engine_get_s8(path, &value);
+		if (ret == 0) {
+			*output = value;
+			ret = 1;
+		}
+	} break;
+	case LWM2M_INT_TYPE_INT16: {
+		s16_t value;
+
+		ret = lwm2m_engine_get_s16(path, &value);
+		if (ret == 0) {
+			memcpy(output, &value, 2);
+			ret = 2;
+		}
+	} break;
+	case LWM2M_INT_TYPE_INT32: {
+		s32_t value;
+
+		ret = lwm2m_engine_get_s32(path, &value);
+		if (ret == 0) {
+			memcpy(output, &value, 4);
+			ret = 4;
+		}
+	} break;
+	case LWM2M_INT_TYPE_INT64: {
+		s64_t value;
+
+		ret = lwm2m_engine_get_s64(path, &value);
+		if (ret == 0) {
+			memcpy(output, &value, 8);
+			ret = 8;
+		}
+	} break;
+	default:
+		break;
+	}
+
+	return ret;
+}
+
+int do_lwm2m_write_int(const u8_t *param, u8_t length)
+{
+	u8_t int_type = *param;
```

Comment:
Would it make sense to `u8_t int_type = *(param++);` here  to avoid the need for `param+1` all below? You could also use `sizeof(value)` for offsets calculation to avoid magic numbers.

Reply:
I understand you well. Please allow me to keep as current to explicitly show how the parameter string is constructed, and the length of each component.